File: sample/src/main/java/com/schibsted/spain/barista/sample/KeyboardActivity.java
package com.schibsted.spain.barista.sample;

import android.os.Bundle;
import androidx.appcompat.app.AppCompatActivity;
import android.view.KeyEvent;
import android.widget.EditText;
import android.widget.TextView;

public class KeyboardActivity extends AppCompatActivity {

  private TextView actionsTextView;

  @Override
  protected void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    setContentView(R.layout.activity_keyboard);

    actionsTextView = (TextView) findViewById(R.id.actions);
    initEditText();
    initEditTextNotFocused();
    initEditTextVeryFarAway();
  }

  private void initEditText() {
    EditText editText = (EditText) findViewById(R.id.edittext);
    editText.setOnEditorActionListener(new TextView.OnEditorActionListener() {
      @Override
      public boolean onEditorAction(TextView textView, int i, KeyEvent keyEvent) {
        actionsTextView.setText("Edit text ime action button pressed!");
        return true;
      }
    });
  }

  private void initEditTextNotFocused() {
    EditText editText = (EditText) findViewById(R.id.edittext_not_focused);
    editText.setOnEditorActionListener(new TextView.OnEditorActionListener() {
      @Override
      public boolean onEditorAction(TextView textView, int i, KeyEvent keyEvent) {
        actionsTextView.setText("Edit text not focused ime action button pressed!");
        return true;
      }
    });
  }

  private void initEditTextVeryFarAway() {
    EditText editText = (EditText) findViewById(R.id.edittext_very_far_away);
    editText.setOnEditorActionListener(new TextView.OnEditorActionListener() {
      @Override
      public boolean onEditorAction(TextView textView, int i, KeyEvent keyEvent) {
        actionsTextView.setText("Edit text very far away ime action button pressed!");
        return true;
      }
    });
  }
}

File: sample/src/main/java/com/schibsted/spain/barista/sample/KeyboardActivity.kt
package com.schibsted.spain.barista.sample

import android.os.Bundle
import android.view.View
import android.widget.EditText
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity

class KeyboardActivity : AppCompatActivity() {
  private var actionsTextView: TextView? = null
  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    setContentView(R.layout.activity_keyboard)
    actionsTextView = findViewById<View>(R.id.actions) as TextView
    initEditText()
    initEditTextNotFocused()
    initEditTextVeryFarAway()
  }

  private fun initEditText() {
    val editText = findViewById<View>(R.id.edittext) as EditText
    editText.setOnEditorActionListener { textView, i, keyEvent ->
      actionsTextView!!.text = "Edit text ime action button pressed!"
      true
    }
  }

  private fun initEditTextNotFocused() {
    val editText = findViewById<View>(R.id.edittext_not_focused) as EditText
    editText.setOnEditorActionListener { textView, i, keyEvent ->
      actionsTextView!!.text = "Edit text not focused ime action button pressed!"
      true
    }
  }

  private fun initEditTextVeryFarAway() {
    val editText = findViewById<View>(R.id.edittext_very_far_away) as EditText
    editText.setOnEditorActionListener { textView, i, keyEvent ->
      actionsTextView!!.text = "Edit text very far away ime action button pressed!"
      true
    }
  }
}